e2e/support/helpers/e2e-notebook-helpers.ts

```diff
@@ -0,0 +1,201 @@
+import { popover } from "e2e/support/helpers/e2e-ui-elements-helpers";
+
+/**
+ * Switch to a notebook editor from a simple query view (aka "chill mode").
+ */
+export function openNotebook() {
+  return cy.findByTestId("qb-header-action-panel").icon("notebook").click();
+}
+
+/**
+ * Select a specific notebook step like filter, join, breakout, etc.
+ *
+ * @see {@link https://github.com/metabase/metabase/pull/17708#discussion_r700082403}
+ */
+export function getNotebookStep(
+  type:
+    | "summarize"
+    | "join"
+    | "expression"
+    | "filter"
+    | "sort"
+    | "limit"
+    | "data",
+  { stage = 0, index = 0 } = {},
+): Cypress.Chainable<JQuery<HTMLElement>> {
+  return cy.findByTestId(`step-${type}-${stage}-${index}`);
+}
+
+/**
+ * @summary Visualize notebook query results.
+ *
+ * This helper intelligently waits for the query to load, and gives you an option
+ * to assert on the waited query response.
+ *
+ * @example
+ * visualize();
+ *
+ * visualize(response => {
+ *   expect(response.body.error).to.not.exist;
+ * });
+ */
+export function visualize(callback?: (response?: object) => void) {
+  cy.intercept("POST", "/api/dataset").as("dataset");
+
+  cy.button("Visualize").click();
+
+  cy.wait("@dataset").then(({ response }) => {
+    if (callback) {
+      callback(response);
+    }
+  });
+}
+
+/**
+ * Summarize (Aggregate).
+ *
+ * Doesn't support summarizing using Custom Expression!
+ */
+export function addSummaryField({
+  metric,
+  table,
+  field,
+  stage = 0,
+  index = 0,
+}: {
+  metric:
+    | "Count of rows"
+    | "Sum of ..."
+    | "Average of ..."
+    | "Number of distinct values of ..."
+    | "Cumulative sum of ..."
+    | "Cumulative count of rows"
+    | "Standard deviation of ..."
+    | "Minimum of ..."
+    | "Maximum of ...";
```

Comment:
1. These aren't all [basic aggregation operators](https://github.com/metabase/metabase/blob/6265ac4fffe4889efa8175cb75fc78d10affaa1a/frontend/src/metabase-lib/operators/constants.js#L347-L355).
2. I don't think we should be adding types based on locale-dependent labels.

```suggestion
  metric: string;
```

Reply:
This is a good example of why types don't necessarily need to be shared between FE and E2E. From my point of view, this helper doesn't care about all aggregation operators at all. On the contrary. The function of these enumerated types is to inform you (consumer) of which possible values you can use when abstracting the user flow with this helper. And those are the strings that you can see in the UI.

The second argument is much stronger, imo. But then again, 99.999% of all our E2E tests are using English. There are maybe one or two tests that are testing different locales specifically.

Reply:
I'd go with a `string` like @kamilmielnik suggests.

Reply:
Done https://github.com/metabase/metabase/pull/36474/commits/a65b52eb0d387400381538d6d2d06559a8e1a017